PowerShell/JumpCloud Module/Public/Users/Remove-JCUser.ps1

```diff
@@ -26,39 +26,64 @@ UserID has an Alias of _id. This means you can leverage the PowerShell pipeline
         [Switch]
         $ByID,
-
+        # Do not use $CascadeManager if $force is used
         [Parameter(HelpMessage = 'A SwitchParameter which suppresses the warning message when removing a JumpCloud User.')]
         [Switch]
-        $force
+        $force,
+        [Parameter(HelpMessage = 'A SwitchParameter for Cascading the manager of the user to the users managed by the user. NULL, AUTOMATIC (bubble up), ID (prompt for manager ID)')]
+        [ValidateSet('NULL', 'Auto', 'ID')]
+        [string]$CascadeManager
     )
-
+    DynamicParam {
+        # Create a dynamic parameter to get the -CascadeManagerId
+        if ($CascadeManager -eq 'ID') {
+            $paramDictionary = New-Object -Type System.Management.Automation.RuntimeDefinedParameterDictionary
+            $paramAttributesCollect = New-Object -Type System.Collections.ObjectModel.Collection[System.Attribute]
+
+            $paramAttributes = New-Object -Type System.Management.Automation.ParameterAttribute
+            $paramAttributes.Mandatory = $true
+            $paramAttributesCollect.Add($paramAttributes)
+
+            $dynParam1 = New-Object -Type System.Management.Automation.RuntimeDefinedParameter("CascadeManagerId", [string], $paramAttributesCollect)
+            $paramDictionary.Add('CascadeManagerId', $dynParam1)
+            return $paramDictionary
+        }
+    }
     begin {
         Write-Debug 'Verifying JCAPI Key'
         if ([System.String]::IsNullOrEmpty($JCAPIKEY)) {
             Connect-JConline
         }
-
         Write-Debug 'Populating API headers'
         $hdrs = @{
-
             'Content-Type' = 'application/json'
             'Accept'       = 'application/json'
             'X-API-KEY'    = $JCAPIKEY
-
         }
 
         if ($JCOrgID) {
             $hdrs.Add('x-org-id', "$($JCOrgID)")
         }
 
         $deletedArray = @()
-
-        if ($PSCmdlet.ParameterSetName -eq 'Username' ) {
-            $UserHash = Get-DynamicHash -Object User -returnProperties username
-            $UserCount = ($UserHash).Count
-            Write-Debug "Populated UserHash with $UserCount users"
+        # If $cascadeManager and $force are used, throw an error
+        if ($CascadeManager -and $force) {
+            Throw "Cannot use -CascadeManager and -Force together. Please use one or the other."
         }
 
+        $UserHash = Get-DynamicHash -Object User -returnProperties 'username', 'manager'
+        # Validate dynamic parameter
+        if ($CascadeManager -eq 'ID') {
+            $CascadeManagerId = $PSBoundParameters['CascadeManagerId']
+            # Validate if the Id is a JC User from the $UserHash
+            if ($UserHash.ContainsKey($CascadeManagerId)) {
+                Write-Debug "CascadeManagerId is a valid JumpCloud User"
+            } else {
+                Write-Error "User does not exist. Please enter a valid UserID."
+                # Exit the script
+                Exit
```

Comment:
Why would we exit here, this doesn't make sense? 

Try to run your code 
`Remove-JCUser -username "jeff.fisher" -CascadeManager id -CascadeManagerId bob` with a managerID that doesn't exist. Tell me if exit is the right choice here.

In addition, we should have a test for this.

Reply:
Yeah I believe throw would be more appropriate in this scenario